src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -152,19 +152,35 @@ def __init__(
         # Siblings are other dataframes that share the same query compiler. We
         # use this list to update inplace when there is a shallow copy.
         from snowflake.snowpark.modin.pandas.utils import try_convert_index_to_native
+        from snowflake.snowpark.modin.plugin.extensions.index import Index
 
         self._siblings = []
 
+        if isinstance(index, DataFrame):  # pandas raises the same error
```

Comment:
@sfc-gh-joshi what's the best way to update this? Do we need an override file?

Reply:
Yes, I think we'd have to use the extensions to override __init__. I'm working on a PR on the modin side that would let us override a separate method when a modin DF/Series or query compiler is passed to the constructor.